src/main/java/org/folio/entitlement/integration/token/TokenProvider.java

```diff
@@ -0,0 +1,47 @@
+package org.folio.entitlement.integration.token;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.folio.entitlement.integration.keycloak.KeycloakCacheableService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+/**
+ * Provides access tokens for internal operations between mgr-* services.
+ * Uses KeycloakCacheableService to obtain and cache tokens, ensuring fresh tokens
+ * are available even during long-running operations.
+ */
+@Log4j2
+@Component
+@RequiredArgsConstructor
+public class TokenProvider {
+
+  private KeycloakCacheableService keycloakCacheableService;
+
+  @Value("${application.keycloak.enabled}")
+  private boolean keycloakEnabled;
+
+  @Autowired(required = false)
```

Comment:
Since the behavior controlled by application.keycloak.enabled configuration property it's better to have to different implementations (Spring beans): one always returns user token (Keycloak is disabled), another -- gets fresh token from KeycloakCacheableService 

This way you won't be needed to place "if" condition into the getToken() method and inject non-required dependency (KeycloakCacheableService )